framework/SupervisedLearning/KerasLSTMRegression.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Created on Apr. 13, 2021
+
+  @author: cogljj
+  module for recurrent neural network using short-term model network (LSTM)
+"""
+#External Modules------------------------------------------------------------------------------------
+import numpy as np
+######
+#Internal Modules------------------------------------------------------------------------------------
+from .KerasRegression import KerasRegression
+#Internal Modules End--------------------------------------------------------------------------------
+
+class KerasLSTMRegression(KerasRegression):
+  """
+    recurrent neural network using short-term model network (LSTM) regression
+    constructed using Keras API in TensorFlow
+  """
+
+  def __init__(self, **kwargs):
+    """
+      A constructor that will appropriately intialize a supervised learning object
+      @ In, kwargs, dict, an arbitrary dictionary of keywords and values
+      @ Out, None
+    """
+    KerasRegression.__init__(self, **kwargs)
+    self.printTag = 'KerasLSTMRegression'
+    self.allowedLayers = self.basicLayers + self.kerasROMDict['kerasRcurrentLayersList']
+
+  def _checkLayers(self):
+    """
+      Method used to check layers setups for KERAS model
+      @ In, None
+      @ Out, None
+    """
+    for index, layerName in enumerate(self.layerLayout[:-1]):
+      layerType = self.initOptionDict[layerName].get('type').lower()
+      if layerType in ['lstm']:
+        if not self.initOptionDict[layerName].get('return_sequences'):
+          self.initOptionDict[layerName]['return_sequences'] = True
+          self.raiseAWarning('return_sequences is resetted to True for layer',layerName)
+
+  def _preprocessInputs(self,featureVals):
```

Comment:
@wangcj05 Hm, I think I could move _preprocessInputs to KerasBase, if you agree that would be a good idea. I think that removes the major redundancy in KerasLSTMRegression and KerasLSTMCLassifier.

Reply:
I think this is tied to LSTM. It is not general all neural network.

Reply:
Hm, then I could just make a LSTM utils, and put the function in that, and use it in both.

Reply:
Tho' since @alfoa is going to be making changes to this soon, maybe I should just put a comment in both files that it is identical, and after the changes we can decide if it is worth changing.

Reply:
To me, a comment is fine. I think my comments are more for future. How do we want the DNN classes inheritance look like? @alfoa any more comments? BTW, the issue that I raised can be further discussed and addressed in future PR.